### ISSUE TITLE
status.conditions.reason: Invalid value: \"null\"

### ISSUE THREAD

**Issue by 5elf** [chaos-mesh/chaos-mesh#3526]
## Bug Report

**What version of Kubernetes are you using?**

Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.1", GitCommit:"5e58841cce77d4bc13713ad2b91fa0d961e69192", GitTreeState:"clean", BuildDate:"2021-05-12T14:18:45Z", GoVersion:"go1.16.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"18+", GitVersion:"v1.18.10-468+23b6e83f628328-dirty", GitCommit:"23b6e83f6283287ee0674953488b478ab6a740f7", GitTreeState:"dirty", BuildDate:"2022-06-13T06:58:27Z", GoVersion:"go1.17.3", Compiler:"gc", Platform:"linux/amd64"}

**What version of Chaos Mesh are you using?**

v2.2.0

**What did you do?**

Create a StressChaos in cluster

```yaml
---
apiVersion: chaos-mesh.org/v1alpha1
kind: StressChaos
metadata:
  name: burn-cpu
spec:
  mode: one
  selector:
    namespaces:
    - flink
    labelSelectors:
      "app.kubernetes.io/instance": "sleep-5elf-chart"
  stressors:
    cpu:
      workers: 1
      load: 100
  duration: "30s"
```

**What did you expect to see?**

The stress-ng processes appeared in the `top` command and exited after 30 seconds.

**What did you see instead?**

The stress-ng started successfully but didn't exit

**Output of chaosctl**

The controller manager log as below
inject phase
```text
2022-08-04T07:55:41.564Z        DEBUG   controller-runtime.webhook.webhooks     admission/http.go:136   wrote response  {"webhook"
: "/mutate-chaos-mesh-org-v1alpha1-stresschaos", "code": 200, "reason": "", "UID": "25bfe43e-46b6-4c8f-8b11-7c8ad3c72745", "allowe
d": true}
2022-08-04T07:55:41.566Z        ERROR   records records/controller.go:220       fail to update  {"error": "StressChaos.chaos-mesh.org \"burn-cpu\" is invalid: status.instances.flink/sleep-5elf-chart-647b68dcc4-dlqsp/sleep-5elf-chart.memoryUid: Invalid value: \"null\": status.instances.flink/sleep-5elf-chart-647b68dcc4-dlqsp/sleep-5elf-chart.memoryUid in body must be of type string: \"null\""}
github.com/chaos-mesh/chaos-mesh/controllers/common/records.(*Reconciler).Reconcile
        /home/runner/work/chaos-mesh/chaos-mesh/controllers/common/records/controller.go:220
github.com/chaos-mesh/chaos-mesh/controllers/common/pipeline.(*Pipeline).Reconcile
        /home/runner/work/chaos-mesh/chaos-mesh/controllers/common/pipeline/pipeline.go:73
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile
        /tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:114
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
        /tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:311
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
        /tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:266
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
        /tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:227
2022-08-04T07:55:41.566Z        INFO    pipeline        pipeline/pipeline.go:78 reconcile result        {"result": {"Requeue":true,"RequeueAfter":0}}
2022-08-04T07:55:41.571Z        INFO    pipeline        pipeline/pipeline.go:78 reconcile result        {"result": {"Requeue":false,"RequeueAfter":0}}
2022-08-04T07:55:41.571Z        INFO    desiredphase    desiredphase/controller.go:129  modify desiredPhase     {"desiredPhase": "Run"}
2022-08-04T07:55:41.571Z        INFO    pipeline        pipeline/pipeline.go:78 reconcile result        {"result": {"Requeue":false,"RequeueAfter":29428283578}}
2022-08-04T07:55:41.571Z        INFO    condition       condition/controller.go:122     updating conditions     {"conditions": [{"type":"Paused","status":"False","reason":""},{"type":"Selected","status":"False","reason":""},{"type":"AllInjected","status":"True","reason":""},{"type":"AllRecovered","status":"True","reason":""}]}
```

recover phase:
```text
2022-08-04T07:56:22.956Z        ERROR   condition       condition/controller.go:138     fail to update  {"error": "StressChaos.chaos-mesh.org \"burn-cpu\" is invalid: status.conditions.reason: Invalid value: \"null\": status.conditions.reason in body must be of type string: \"null\""}
github.com/chaos-mesh/chaos-mesh/controllers/common/condition.(*Reconciler).Reconcile
        /home/runner/work/chaos-mesh/chaos-mesh/controllers/common/condition/controller.go:138
github.com/chaos-mesh/chaos-mesh/controllers/common/pipeline.(*Pipeline).Reconcile
        /home/runner/work/chaos-mesh/chaos-mesh/controllers/common/pipeline/pipeline.go:73
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile
        /tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:114
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
        /tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:311
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
        /tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:266
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
        /tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:227
2022-08-04T07:56:22.956Z        INFO    pipeline        pipeline/pipeline.go:78 reconcile result        {"result": {"Requeue":false,"RequeueAfter":0}}
2022-08-04T07:56:22.956Z        INFO    records records/controller.go:117       iterating record        {"record": {"id":"flink/sleep-5elf-chart-647b68dcc4-dlqsp/sleep-5elf-chart","selectorKey":".","phase":"Not Injected"}, "desiredPhase": "Stop"}
2022-08-04T07:56:22.956Z        INFO    pipeline        pipeline/pipeline.go:78 reconcile result        {"result": {"Requeue":false,"RequeueAfter":0}}
2022-08-04T07:56:22.956Z        DEBUG   events  recorder/recorder.go:103        Normal  {"object": {"kind":"StressChaos","namespace":"flink","name":"burn-cpu","uid":"55dd284a-6dca-4e3e-9de3-62f931390022","apiVersion":"chaos-mesh.org/v1alpha1","resourceVersion":"154517368"}, "reason": "Failed", "message": "Failed to update conditions: StressChaos.chaos-mesh.org \"burn-cpu\" is invalid: status.conditions.reason: Invalid value: \"null\": status.conditions.reason in body must be of type string: \"null\""}
2022-08-04T07:56:22.956Z        DEBUG   events  recorder/recorder.go:103        Normal  {"object": {"kind":"StressChaos","namespace":"flink","name":"burn-cpu","uid":"55dd284a-6dca-4e3e-9de3-62f931390022","apiVersion":"chaos-mesh.org/v1alpha1","resourceVersion":"154517368"}, "reason": "Failed", "message": "Failed to update conditions: StressChaos.chaos-mesh.org \"burn-cpu\" is invalid: status.conditions.reason: Invalid value: \"null\": status.conditions.reason in body must be of type string: \"null\""}
```

**Comment by cwen0** [comment]
@5elf How did you install Chaos Mesh v2.2.0? I guess that the CRD version is old.

**Comment by 5elf** [comment]
> @5elf How did you install Chaos Mesh v2.2.0? I guess that the CRD version is old.

I think the CRDs are okay.


I got this error message because the checks and validations are more strict in our internal version of the Kubernetes cluster. And the problem was gone after I added some missing `omitempty` JSON tags.

I think it will be better if we could explicitly specify that these optional fields can be omitted.

I will create a PR to solve it soon.
